includes/Blocks/Block.php

```diff
@@ -292,10 +292,18 @@ private function create_attributes_fields( $attributes, $prefix ): array {
 	 * @param array|string $value The value
 	 * @param string       $type The type of the value
 	 * 
-	 * @return mixed
+	 * @return array|string|int|float|bool
 	 */
 	private function normalize_attribute_value( $value, $type ) {
+		// @todo use the `source` to normalize array/object values.
+		if ( is_array( $value ) ) {
+			return $value;
+		}
```

Comment:
I think it would be clearer to move this into the `array` case block, and if $type is _not_ array, we shouldn't return an array, right?